.github/workflows/ci.yml

```
@@ -73,9 +78,99 @@ jobs:
+          lcov --capture --directory build --output-file coverage.info --ignore-errors mismatch,gcov
+          lcov --remove coverage.info \
+            '/usr/*' \
+            '*/external/*' \
+            '*/tests/*' \
+            '*/build/*' \
+            --output-file coverage.filtered.info \
+            --ignore-errors unused
+          genhtml coverage.filtered.info --output-directory coverage-report
+          lcov --summary coverage.filtered.info | tee coverage-summary.txt
+
+      -      - name: Enforce Coverage Threshold
```

YAMLlint (1.38.0)

[warning] too many spaces after hyphen

(hyphens)